java/src/main/java/com/saucelabs/simplesauce/SauceSession.java

```diff
@@ -223,42 +223,12 @@ public SauceSession withMacOsXYosemite() {
         browserName = "Safari";
         return this;
     }
-    //TODO notice the duplication below with edge.
+    //FIXED notice the duplication below with edge. Used enumeration
     //Maybe could be moved to a separate class so we can do withEdge().16_16299();
     //Or withEdge().version(EdgeVersion.16_16299);
-    public SauceSession withEdge16_16299() {
+    public SauceSession withEdge(EdgeVersion version) {
```

Comment:
That's an interesting solution. I'd like to get your feedback on this:
So, the idea here with these bindings is to have as much static typing as possible. Meaning, we wanted to do stuff like: `withEdge16()` and `withEdge17()` and so on. That way, the user isn't left guessing about what all of the supported Edge versions are.
Your solution solves the same problem a different way, using an `enum`. 
[Here](https://github.com/saucelabs/simple_sauce/pull/43/files#diff-737dcd8dd9072c63fbbd35c93414eabfR102-R111) I actually fixed the naming convention of the Edge versions to `withEdge15()` rather than 'withEdge15_5757'. I tend to like the new name better. In which case, using an Enum will not work.
What are your thoughts?

Reply:
Personally I was thinking Enums could be helpful somewhere here. We might be able to default to a version too with an overloaded method.

Reply:
I've updated my pull request to update the naming convention as per nadvolod's suggestions. You can view the updates [here](https://github.com/saucelabs/simple_sauce/pull/47/commits/3168ce4295b7b4219081fd885afdbe7ef3900328). The request still uses enumerations, but changes the version name.

Reply:
@joshmgrant hey bud, can you elaborate? Are you liking the direction that @jaeheonshim is going you're saying? Or something else?